File: examples/awfy/node/list.mjs
export const warmup = 10, iters = 40, size = 1000, expected = 499500;

function build(n) {
  let acc = null;
  for (let i = 0; i < n; i++) acc = { v: i, next: acc };
  return acc;
}

function sumList(xs) {
  let acc = 0;
  for (let cur = xs; cur !== null; cur = cur.next) acc += cur.v;
  return acc;
}

export function run(size) {
  const xs = build(size);
  return sumList(xs);
}

File: examples/performance/awfy/go/list.go
package main

type listNode struct {
	v    int
	next *listNode
}

func listBuild(n int) *listNode {
	var acc *listNode
	for i := 0; i < n; i++ {
		acc = &listNode{v: i, next: acc}
	}
	return acc
}

func listSum(xs *listNode) int {
	acc := 0
	for cur := xs; cur != nil; cur = cur.next {
		acc += cur.v
	}
	return acc
}

func listRun(size int) int {
	return listSum(listBuild(size))
}

var listBench = Bench{Name: "list", Warmup: 10, Iters: 40, Size: 1000, Expected: 499500, Run: listRun}
